addons/font/font.c

```diff
@@ -189,6 +189,36 @@ static int color_render_char(const ALLEGRO_FONT* f,
    return w;
 }
 
+
+/* color_render_char_scaled:
+ *  (color vtable entry)
+ *  Renders a color character onto a bitmap, at the specified location,
+ *  using the specified color, scaled.
+ *  Returns the character width, in pixels.
+ */
+static int color_render_char_scaled(const ALLEGRO_FONT* f,
+	ALLEGRO_COLOR color, int ch, float x, float y, float sw, float sh)
+{
+	int w = 0;
+	int h = f->vtable->font_height(f);
+	ALLEGRO_BITMAP *g;
+
+	g = _al_font_color_find_glyph(f, ch);
+	if (g) {
+		int width = al_get_bitmap_width(g);
+		int height = al_get_bitmap_height(g);
+		w = width * sw;
+		al_draw_tinted_scaled_bitmap(g, color, 0, 0, width, height,
+			x, y + ((float)h - height) / 2.0f, w, height * sh, 0);
```

Comment:
I believe that the vertical placement is incorrect here.  The `height` should have the scaling applied.  I suggest:
`y + ((float)h - height * sh) / 2.0f`